src/components/ListFilterBar/ListFilterBar.js

```diff
@@ -0,0 +1,164 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Icon, Button, Dropdown, Input, Menu} from 'semantic-ui-react';
+import Badge from '../Badge/Badge';
+import {
+  fileLatestStatus,
+  versionState,
+  fileTypeDetail,
+  createDateSort,
+  modifiedDateSort,
+  defaultSort,
+} from '../../common/fileUtils';
+
+/**
+ * Filter Bar for Study Files, returns filtered list in "filteredList" render prop
+ */
+const ListFilterBar = ({fileList, filteredList}) => {
+  const [sortMethod, setSortMethod] = useState('');
+  const [sortDirection, setSortDirection] = useState('ascending');
+  const [typeFilterStatus, setTypeFilterStatus] = useState('');
+  const [approvalFilterStatus, setApprovalFilterStatus] = useState('');
+  const [searchString, setSearchString] = useState('');
+
+  const statusOptions = Object.keys(versionState).map(state => ({
+    key: state,
+    value: state,
+    text: versionState[state].title,
+    content: <Badge state={state} />,
+  }));
+
+  const typeOptions = Object.keys(fileTypeDetail).map(type => ({
+    key: type,
+    value: type,
+    text: fileTypeDetail[type].title,
+    content: (
+      <small>
+        <Icon name={`${fileTypeDetail[type].icon || 'question'}`} />
+        {' ' + fileTypeDetail[type].title}
+      </small>
+    ),
+  }));
+
+  const sortOptions = [
+    {
+      key: 'createDate',
+      value: 'createDate',
+      text: 'Create date',
+    },
+    {
+      key: 'modifyDate',
+      value: 'modifyDate',
+      text: 'Modified date',
+    },
+  ];
+
+  const sortedFileList = () => {
+    const sortFuncs = {
+      createdDate: createDateSort,
+      modifyDate: modifiedDateSort,
+      default: defaultSort,
+    };
+    var sortedList = fileList.sort(sortFuncs[sortMethod] || sortFuncs.default);
+    sortedList =
+      sortDirection === 'ascending' ? sortedList : sortedList.reverse();
+    sortedList = sortedList.filter(obj =>
+      obj.node.fileType.includes(typeFilterStatus),
+    );
+    sortedList = sortedList.filter(obj =>
+      fileLatestStatus(obj.node).includes(approvalFilterStatus),
+    );
+    sortedList = sortedList.filter(
+      obj =>
+        obj.node.name.toLowerCase().includes(searchString.toLowerCase()) ||
+        obj.node.description.toLowerCase().includes(searchString.toLowerCase()),
+    );
+    return sortedList;
+  };
+
+  return (
+    <>
+      <Menu size="small">
```

Comment:
Did you consider using a form group with inline fields here? It may help with some gutters and making it more consistent with other input fields.